### PR TITLE
[RFC] adopt contributor covenant v2

### PR DESCRIPTION
**Pull request by achrinza**
This PR consists of 1 commit:

- Adopt Contributor Covenant v2

    This is to steer the project to be more in-line with the expectations from the FOSS community, thereby increasing ease of adoption and contribution.

    Electron, AMP, nvm, Node-RED are examples of Node.js projects that adopt the Contributor Covenant.

Signed-off-by: Rifa Achrinza <25147899+achrinza@users.noreply.github.com>

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- API Documentation in code was updated
- Documentation in [/docs/site](../tree/master/docs/site) was updated
- Affected artifact templates in `packages/cli` were updated
- Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by bajtos** [comment]
Thank you @achrinza for the pull request. I don't have strong opinions on details about which form of code of conduct to adopt. @dhmlau @raymondfeng what's your take? Are we ok to adopt "Contributor Covenant v2"? Do we need to get any IBM-internal approvals for this change?

File: docs/CODE_OF_CONDUCT.md
Comment (bajtos) [inline docs/CODE_OF_CONDUCT.md]: Please replace this naked URL with a `mailto:` link.

File: docs/CONTRIBUTING.md
Comment (bajtos) [inline docs/CONTRIBUTING.md]: I think we will need to update the documentation in http://loopback.io/doc/en/contrib/index.html too.

**Comment by emonddr** [comment]
@raymondfeng , @dhmlau , your opinion on this? thanks. :)

**Comment by dhmlau** [comment]
I need to find out what's the implications and recommendation from the IBM open source side.  Will circle back.

**Comment by dhmlau** [comment]
We just switch the contribution method from CLA to DCO, making your contribution easier in the future. Please sign the commits with DCO by amending your commit messages with `-s` flag and push the changes again. If you're using the command line, you can:

```
git commit --amend -s
git push --force-with-lease
```

Please refer to [this docs page](https://loopback.io/doc/en/contrib/code-contrib.html#developer-certificate-of-origin-dco) for details. Thanks!

**Comment by achrinza** [comment]
@dhmlau Any updates on your front? This PR is ready to go, just needs approval 😀

**Review by bajtos — APPROVED** [review]
LGTM as far as I am concerned. Please wait for an approval from @dhmlau and @raymondfeng before landing.

Thank you @achrinza for initiating this change!

**Review by raymondfeng — APPROVED** [review]

**Review by dhmlau — APPROVED** [review]

**Comment by dhmlau** [comment]
@achrinza, could you please take a look at the commit linter issue? Thanks.
```
✖   body's lines must not be longer than 100 characters [body-max-line-length]
```

**Comment by bajtos** [comment]
Landed, thank you @achrinza again!

Can you please take a look at the contrib docs on loopback.io (start with http://loopback.io/doc/en/contrib/index.html) and check what needs to be updated too?

I think we should add a link to CODE OF CONDUCT to Governance Principles - see https://loopback.io/doc/en/contrib/Governance.html#principles.

**Pull request merged**
